### PR TITLE
Adding a Makefile for local development and build process

### PR DESCRIPTION
**Pull request by glower**
Currently we have different scripts for building, testing and running the service spread over different places. This Makefile should combine everything in one place.

* Added a Makefile to build, run and tests the service
* Release process doesn't require to add the new version to config.go anymore. The correct version and revision will be added automatically based on the latests tag and commit hash
* Added additional script to bootstrap lxd container
* Renamed `settings.yaml` to `settings.yaml.example` and added settings.yaml to `.gitignore` file. So the developer can have a modified file without committing it

### CODE REVIEW

**Comment by glower** [comment]
After the Makefile is merged I will update documentation and all the existing scripts and config files to only use Makefile targets

**Comment by doismellburning** [comment]
@glower Was that merge you missing my comments above or deciding that it wasn't necessary? Would it have helped if I'd used GitHub's review functionality?

**Comment by glower** [comment]
> @glower Was that merge you missing my comments above or deciding that it wasn't necessary? Would it have helped if I'd used GitHub's review functionality?

Sorry, I forgot to replay you. I decided to update the documentation and the automation in a separate commit

File: dependencies.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+postgresql-12
```
[the remark this answers is not in the document]
Comment (glower) [reply dependencies.txt:1]: Yes, I would add a entry in the documentation about how to run/migrate the service to Focal

**Comment by glower** [comment]
> The other thing I'm very conscious of here: I don't see updates to the documentation or to any automated tests.

I am currently testing if it would make sense to only use github actions with the new Makefile. So we don't have two different systems for automation. After this is done and working the documentation will be updated.

**Comment by doismellburning** [comment]
> I decided to update the documentation and the automation in a separate commit

Separate commit, fine, but in future I'd like to see documentation and tests in *the same PR* please - they're part of something being Done, and that way we're much more likely to ensure they actually happen.

**Pull request merged**